dask_jobqueue/core.py

```diff
@@ -14,17 +19,119 @@ class JobQueueCluster(object):
     This class should not be used directly, use inherited class appropriate
     for your queueing system (e.g. PBScluster or SLURMCluster)
 
+    Abstract init parameters
+    ------------------------
+    name : str
+        Name of Dask workers.
+    threads : int
+        Number of threads per process.
+    processes : int
+        Number of processes per node.
+    memory : str
+        Bytes of memory that the worker can use. This should be a string
+        like "7GB" that can be interpretted both by PBS and Dask.
+    interface : str
+        Network interface like 'eth0' or 'ib0'.
+    death_timeout : float
+        Seconds to wait for a scheduler before closing workers
+    local_directory : str
+        Dask worker local directory for file spilling.
+    extra : str
+        Additional arguments to pass to `dask-worker`
+    kwargs : dict
+        Additional keyword arguments to pass to `LocalCluster`
+
     See Also
     --------
     PBSCluster
     SLURMCluster
     """
-    def __init__(self):
-        raise NotImplemented
+
+    _script_template ="""
+#!/bin/bash
+
+%(job_header)s
+
+%(worker_command)s 
+""".lstrip()
+
+    def __init__(self,
+                 name='dask-worker',
+                 threads=4,
+                 processes=6,
+                 memory='16GB',
+                 interface=None,
+                 death_timeout=60,
+                 local_directory=None,
+                 extra='',
+                 **kwargs
+                 ):
+        """
+        This initializer should be considered as Abstract, and never used directly.
+        :param name:
+        :param threads:
+        :param processes:
+        :param memory:
+        :param interface:
+        :param death_timeout:
+        :param local_directory:
+        :param extra:
+        :param kwargs:
```

Comment:
So far we've been using numpydoc style docstrings: http://dask.pydata.org/en/latest/develop.html#docstrings